pkg/solana/chainreader/chain_reader.go

```diff
@@ -15,44 +16,59 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
 
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logpoller"
 )
 
-const ServiceName = "SolanaChainReader"
+type EventsReader interface {
+	RegisterFilter(context.Context, logpoller.Filter) error
+	FilteredLogs(context.Context, []query.Expression, query.LimitAndSort, string) ([]logpoller.Log, error)
+}
+
+const ServiceName = "SolanaContractReader"
 
-type SolanaChainReaderService struct {
+type ContractReaderService struct {
 	types.UnimplementedContractReader
 
-	// provided values
+	// provided dependencies
 	lggr   logger.Logger
 	client MultipleAccountGetter
+	events EventsReader
 
 	// internal values
 	bindings namespaceBindings
 	lookup   *lookup
 	parsed   *codec.ParsedTypes
 	codec    types.RemoteCodec
+	filters  []logpoller.Filter
 
 	// service state management
 	wg sync.WaitGroup
 	services.StateMachine
 }
 
 var (
-	_ services.Service     = &SolanaChainReaderService{}
-	_ types.ContractReader = &SolanaChainReaderService{}
+	_ services.Service     = &ContractReaderService{}
+	_ types.ContractReader = &ContractReaderService{}
 )
 
-// NewChainReaderService is a constructor for a new ChainReaderService for Solana. Returns a nil service on error.
-func NewChainReaderService(lggr logger.Logger, dataReader MultipleAccountGetter, cfg config.ContractReader) (*SolanaChainReaderService, error) {
-	svc := &SolanaChainReaderService{
+// NewContractReaderService is a constructor for a new ContractReaderService for Solana. Returns a nil service on error.
+func NewContractReaderService(
+	lggr logger.Logger,
+	dataReader MultipleAccountGetter,
+	cfg config.ContractReader,
+	events EventsReader,
+) (*ContractReaderService, error) {
+	svc := &ContractReaderService{
 		lggr:     logger.Named(lggr, ServiceName),
 		client:   dataReader,
 		bindings: namespaceBindings{},
 		lookup:   newLookup(),
 		parsed:   &codec.ParsedTypes{EncoderDefs: map[string]codec.Entry{}, DecoderDefs: map[string]codec.Entry{}},
+		filters:  []logpoller.Filter{},
```

Comment:
Just noticed the initialization of `events` field is missing here:
```suggestion
		filters:  []logpoller.Filter{},
		events: events,
```
Also, reminder... weren't we going to rename this?